### PR TITLE
Register routes without parameters first

### LINKED ISSUE #35: Route paths must be sorted after definition and before adding them to express.
#### Problem

Routes must follow a precedence definition when adding them to express.

```
app.get('things/:id', func);
app.get('things/dance', func);
```

The second api will never match because the dynamic `id` key will match `dance`. So in order to solve this, the routes must be defined as such:

```
app.get('things/dance', func);
app.get('things/:id', func);
```

#### Proposal

Maybe we should automatically sort the routes by their path definition, giving precedence to **not** dynamic path keys.

##### Pros
- avoid user mistake when defining routes

##### Cons
- doing this automatically could be an issue if later on the road we allowed `Route#path` to be a function instead of string.

### PR DESCRIPTION
**Pull request by nolleto**
Closes #35

### CODE REVIEW

File: source/routes.ts [hunk 1]
```diff
@@ -67,6 +67,18 @@ function mergeRoutesWithGlobalOverrides(
   return routes;
 }
 
+function getNumberOfParameter(path: string) {
```
Comment (fsmaia) [inline source/routes.ts:70]: How about `getRouteParametersCount`?

File: source/routes.ts [hunk 2]
```diff
@@ -67,6 +67,18 @@ function mergeRoutesWithGlobalOverrides(
...
+  return (path.match(/:/g) ?? []).length;
+}
+
+function sortRoutesByNumberOfParameters(routes: RouteProperties[]) {
```
Comment (fsmaia) [inline source/routes.ts:74]: And then `sortRoutesByParametersCount`

File: source/routes.ts [hunk 3]
```diff
@@ -98,12 +110,13 @@ export class RouteManager {
       routes,
       this.globalOverrides
     );
+    const routesSorted = sortRoutesByNumberOfParameters(routesWithOverrides);
```
Comment (fsmaia) [inline source/routes.ts:113]: How about `sortedRoutes`?

**Comment by rhberro** [comment]
Hey @nolleto, can you please fix the conflicts so we can merge this one?

**Comment by rhberro** [comment]
I'm closing it until we resolve the conflicts.

**Pull request closed without merging**